File: NameVoting.java
/*
 * Small programm to combine syllables, where some syllables
 * are used only as suffix/prefix
 * 1. Install java SDK
 * 3. Run: javac NameVoting.java && java NameVoting '[Pre "neo", Suf "master", Pre "turbo", Uni "leo", Suf "3000"]'
 **/
 
import java.lang.String;
import java.util.List;
import java.util.LinkedList;
public class NameVoting {


  public static void main(String[] args) {
    System.out.println(NameVoting.combine(args[0]));
  }

  public static List<String> combine(String input) {
    List<String> ps = new LinkedList<String>();
    List<String> ss = new LinkedList<String>();

    for (String word : input.replaceAll("^\\[|\\]$", "").split(",\\s*")) {
      String part[] = word.split(" ") ;
      String val = part[1].replaceAll("^\"|\"$", "");
      if (part[0].equals("Pre")) {
        ps.add(val);
      } else if (part[0].equals("Suf")) {
        ss.add(val);
      } else {
        ps.add(val);
        ss.add(val);
      }
    }

    List<String> result = new LinkedList<String>();
    for (String p : ps)
      for (String s : ss) {
        if (! p.equals(s)) {
          result.add(p + s);
        }
      }
    return result;
  }

}

File: namevoting.rb
# Small programm to combine syllables, where some syllables
# are used only as suffix/prefix
# 1. Install ruby
# 3. Run: ruby namevoting.rb '[Pre "neo", Suf "master", Pre "turbo", Uni "leo", Suf "3000"]'
 
class NameVote
  def self.combine words
    ps = []
    ss = []
    words[1..-2].split(/,\s*/).each do |word| 
      wt = word.split()
      val = wt[1][1..-2]
      case wt[0]
      when "Pre"
        ps << val
      when "Suf"
        ss << val
      else
        ps << val
        ss << val
      end
    end
    ps.product(ss).reject{|x| x[0] == x[1]}.map{|x| x.reduce(:+) }
  end
end

puts (NameVote.combine ARGV[0]).inspect
